ui/src/pages/Results.js

```diff
@@ -0,0 +1,191 @@
+    this.goToAdd = this.goToAdd.bind(this);
+    this.loadResults = this.loadResults.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+  }
+
+  goToAdd() {
+    this.props.history.push({
+          pathname: '/' + this.props.match.params.repoId + '/create',
+      });
+  }
+
+  loadResults() {
+    const query = new URL(window.location.href).searchParams.get('query_name');
+    var apiCalls = [
+        '/' + this.props.match.params.repoId + '/d/repoinfo',
+        '/' + this.props.match.params.repoId + '/d/results?query=' + query,
+        ];
+    Promise.all(apiCalls.map(url => fetch(url)))
+        .then(res => Promise.all(res.map(r => r.json())))
+        .then(
+          (res) => {
+            this.setState({
+              isLoaded: true,
+              repo: res[0],
+              results: res[1]
+            });
+          },
+          (error) => {
+            this.setState({error: error});
+          }
+        );
+  }
+
+  componentDidMount() {
+    this.loadResults();
+  }
+
+  handleSearch(query) {
+    this.setState({isLoaded: false});
+    this.props.history.push({
+        pathname: '/' + this.props.match.params.repoId + '/results',
+        search: '?query_name=' + query,
+      });
+    this.loadResults();
+  }
+
+  renderResults() {
+    if (this.state.results.length == 0) {
+      return (
+        <p className='mdc-typography--body1'>
+          <FormattedMessage {...messages.noResultsFound} />
+        </p>
+      );
+    }
+    var results = this.state.results.map(result => (
+      <SearchResult
+        key={result.personId}
+        repo={this.state.repo}
+        result={result}
+      />
+    ));
+    return <ul className='results-list'>{results}</ul>;
+  }
+
+  renderAddPersonFab() {
+    return (
+      <Fab
+          className='results-addpersonfab'
+          onClick={this.goToAdd}
+          icon={<img src='/static/icons/maticon_add.svg' />}
+          textLabel={this.props.intl.formatMessage(
+              messages.createNewRecord)} />
+    );
+  }
+
+  render() {
+    if (this.state.error) {
+      return <div>An error occurred</div>
+    }
+    if (!this.state.isLoaded) {
+      return <LoadingIndicator />;
+    }
+    return (
+      <div id='results-wrapper'>
+        <RepoHeader
+          repo={this.state.repo}
+          backButtonTarget={'/' + this.state.repo.repoId}
+        />
+        <div className='results-body'>
+          <SearchBar
+              repoId={this.props.match.params.repoId}
+              initialValue={
+                  new URL(window.location.href).searchParams.get('query_name')}
+              onSearch={this.handleSearch} />
+          {this.renderResults()}
+          <Footer />
+        </div>
+        {this.renderAddPersonFab()}
+      </div>);
+  }
+}
+
+/**
+ * A component for an individual search result.
+ */
+class SearchResultImpl extends Component {
+  constructor(props) {
+    super(props);
+    this.goTo = this.goTo.bind(this);
+  }
+
+  goTo() {
```

Comment:
Looks a bit weird that a method named `goTo` takes no parameters. Rename to e.g., `goToView`?